package/gluon-state-check/luasrc/usr/sbin/gluon-state-check

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/lua
+
+local bit = require 'bit'
+local util = require 'gluon.util'
+local unistd = require 'posix.unistd'
+local stat = require 'posix.sys.stat'
+
+local state_dir = "/var/gluon/state/"
+local check_dir = "/lib/gluon/state/check.d/"
+
+
+local function set_flag(stateflag, state)
+	if state then
+		-- this does not modify atime
+		local flaghandle = io.open(stateflag, "w")
+		flaghandle:close()
+	else
+		os.remove(stateflag)
+	end
+end
+
+local function exec_check(checkpath)
+	local checkname = string.sub(checkpath, #check_dir+1)
+	local ret = os.execute(checkpath)
+	local flagfile = state_dir..checkname
+	set_flag(flagfile, 0==ret)
+end
+
+local function run_executable_checks()
+	for _, v in ipairs(util.glob(check_dir..'*')) do
+		local statmode = stat.stat(v).st_mode
+		local is_file = statmode ~= nil and stat.S_ISREG(statmode)
+		if is_file and bit.band(statmode, stat.S_IXUSR) ~= 0 then
```

Comment:
Weird, I thought I had commented on this in my last review, but I can't find the comment now... here's the suggestion again:
```suggestion
		if unistd.access(v, 'x') then
```
(and the `bit` and `stat` requires can be removed then)

Reply:
That's a really nice suggestion, I didn't know `access()` yet.
Though technically I checked for _file_ and _executable_; `access(v, 'x')` tests only for the latter, right?

I'd go with this, following the usage in the [lua-posix](http://luaposix.github.io/luaposix/modules/posix.unistd.html#access) section.
```suggestion
		if unistd.access(v, 'fx') then
```

Reply:
implemented in fa523d6052224188dc6fd2967af1ef0f8283c98f

Reply:
The luaposix docs are a bit inaccurate here, the `access` manpage reveals:

> The value of amode is either the bitwise-inclusive OR of the access permissions to be checked (R_OK, W_OK, X_OK) or the existence test (F_OK).

Meaning that for the Lua version, either a string containing any combination of 'r', 'w' and 'x' can be used, or the value 'f'. Combining 'f' with any other flag doesn't make sense, as 'f' only checks for existence, not for the file type.

So either we keep the `stat`, or we use `unistd.access(v, 'x')`. I don't think we need to care about non-regular files here, so using `access` would be the simplest (and most common) solution.

Reply:
The idea was to give us room to expand later; but I don't mind dropping regular file check if it gets this merged.
resolved in 7f991bf610fc62cb5f5ea9c28e04e36a2c5fa71c